examples/android/CHIPTool/build.gradle

```diff
@@ -1,12 +1,12 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 buildscript {
-    ext.kotlin_version = '1.8.10'
+    ext.kotlin_version = '1.7.21'
     repositories {
         google()
         mavenCentral()
     }
     dependencies {
-        classpath "com.android.tools.build:gradle:4.2.0"
+        classpath "com.android.tools.build:gradle:7.3.0"
```

Comment:
I like this change as a stand-alone (assuming it passes CI): we should update tools versions unless it causes some compatibility issues.

For the version going backwards however (kotlin version above) , I believe a strong justification is needed on why going backward in time is correct.